src/qilib/utils/storage/mongo.py

```diff
@@ -154,46 +162,92 @@ def _retrieve_value_by_tag(self, tag: TagType, parent: ObjectId) -> Any:
+            field: The name of the field to be retrieved
+
+        Returns:
+            The field Data held by the leaf
+
+        """
+        return self._retrieve_value_by_tag(tag, parent, field)
 
-    def _store_value_by_tag(self, tag: TagType, data: Any, parent: ObjectId) -> None:
-        """ Store a value at a given tag
+    def _store_value_by_tag(self, tag: TagType, data: Any, parent: ObjectId, field: Union[str, int, None] = None) -> None:
+        """ Store a value at a given tag. In case a field is specified, function will update the value of the
+        field with data. If the field does not already exists, the field will be created
 
         Args:
             tag: The tag
             data: Data to store
             parent: An ObjectID of the node's parent
+            field: Field to be updated. Default value is None
+
+        Raises:
+              NodeAlreadyExistsError: If a tag in Tag List is an unexpected node/leaf
+              NoDataAtKeyError:  If a tag in Tag List does not exist
         """
 
         if len(tag) == 1:
             doc = self._collection.find_one({'parent': parent, 'tag': tag[0]})
             if doc:
                 if 'value' not in doc:
                     raise NodeAlreadyExistsError(f'Tag "{tag[0]}" is not a leaf')
-                else:
+                elif field is None:
                     self._collection.update_one({'parent': parent, 'tag': tag[0]},
                                                 {'$set': {'value': data}})
-            else:
+                else:
+                    self._collection.update_one({'parent': parent,
+                                                 'tag': tag[0]},
+                                                {'$set': {'value.' +
+                                                          str(field): data}})
+            elif field is None:
                 self._collection.insert_one({'parent': parent, 'tag': tag[0], 'value': data})
-
+            else:
+                raise NoDataAtKeyError(f'Tag "{tag[0]}" does not exist')
         else:
             doc = self._collection.find_one({'parent': parent, 'tag': tag[0]})
             if doc is None:
-                parent = self._collection.insert_one({'parent': parent, 'tag': tag[0]}).inserted_id
+                if field is None:
+                    parent = self._collection.insert_one({'parent': parent, 'tag': tag[0]}).inserted_id
+                else:
+                    raise NoDataAtKeyError(f'Tag "{tag[0]}" does not exist')
             else:
                 if 'value' in doc:
                     raise NodeAlreadyExistsError(f'Tag "{tag[0]}" is a leaf')
                 else:
                     parent = doc['_id']
 
-            self._store_value_by_tag(tag[1:], data, parent)
+            self._store_value_by_tag(tag[1:], data, parent, field)
+
+    def _update_individual_value_by_tag(self, tag: TagType, data: Any, parent: ObjectId, field: Union[str, int, None]) -> None:
```

Comment:
line too long

Reply:
What is the added value of a private method only calling another private method instead of changing all usage of _update_individual_value_by_tag to _store_value_by_tag.